### ISSUE TITLE
Messages fail to be transmitted on application shutdown

### ISSUE THREAD

**Issue by wcom939** [microsoft/ApplicationInsights-Java#1086]
### Expected behavior
On application shutdown, flush is called in order to flush out the remaining messages to send to the cloud.

### Actual behavior
Describe what actually happened that was unexpected.
A race condition occurs whereas while the library attempts to transmit the last of the messages, the httpclient is already stopped causing a "Connection pool shutdown" error in the TransmissionNetworkOutput class.

### To Reproduce
Create a new project, add spring boot, intentionally cause a bean creation error, on your startup application add the flush call in a runtime shutdown hook similar to:
```java
 Runtime.getRuntime().addShutdownHook(new Thread() {
            public void run() {
                try {
                    CustomLogTelemetryClientProxy.flushTelemetry();
                } catch (InterruptedException var2) {
                    var2.printStackTrace();
                }

            }
        });
```
Run the applications a few times without break points, as this is a race condition this isn't always consistent

#### Sample Application
If applicable, provide a sample application which reproduces the issue.

### System information
Please provide the following information:
 - SDK Version: 2.5.0
 - OS type and version: Windows 10
 - Application Server type and version (if applicable): Spring boot
 - Using spring-boot? Yes
 - Additional relevant libraries (with version, if applicable):

### Logs
[Turn on SDK logs](https://docs.microsoft.com/en-us/azure/application-insights/app-insights-java-troubleshoot#debug-data-from-the-sdk) and attach/paste them to the issue. If using an application server, also attach any relevant server logs.

_Be sure to remove any private information from the logs before posting!_

### Screenshots
![image](https://user-images.githubusercontent.com/16446451/66272102-7971fb80-e833-11e9-92ab-37319f661f04.png)

**Comment by asubmani** [comment]
@dhaval24 could you suggest the best way to get this to resolution? We will be opening a support ticket with the name of the AppInsights workspace name.

**Comment by littleaj** [comment]
@wcom939, this is a known issue and is currently in our backlog. I'll update this ticket when we have an ETA.

If your scenario will allow it, you can adjust the [channel settings](https://github.com/microsoft/ApplicationInsights-Java/wiki/ApplicationInsights.XML#channel) (FlushIntervalInSeconds, MaxTelemetryBufferCapacity) to be more aggressive.

**Comment by alitaghavi** [comment]
Any ETA on this?

**Comment by wcom939** [comment]
It does look like this was fixed here https://github.com/microsoft/ApplicationInsights-Java/commit/369eae9837b5c197a217659f6de341a7a5ba1e1a

The previous issue was the client shutdown was not synchronized. I will test the new library and see if it solves the issue

**Comment by trask** [comment]
Resolved in [3.0 Preview](https://docs.microsoft.com/en-us/azure/azure-monitor/app/java-in-process-agent)
